hyper/http11/connection.py

```diff
@@ -75,6 +79,21 @@ def __init__(self, host, port=None, secure=None, ssl_context=None,
         self.ssl_context = ssl_context
         self._sock = None
 
+        # Setup proxy details if applicable.
+        if proxy_host:
+            if proxy_port is None:
+                try:
+                    self.proxy_host, self.proxy_port = proxy_host.split(':')
```

Comment:
This logic could go wrong with IPv6 addresses. I think we might need to pull this out into a helper function, because the logic is a bit tricky, but we can run on some basic heuristics: if the host ends with square bracket it doesn't contain a port. Otherwise, change the split into `rsplit` only allowing one split. That should work in more cases.